plugins/woocommerce/src/Internal/Fulfillments/Fulfillment.php

```diff
@@ -114,8 +114,23 @@ public function set_entity_id( ?string $entity_id ): void {
 	 * @param string|null $status Fulfillment status.
 	 *
 	 * @return void
+	 *
+	 * @throws \InvalidArgumentException If the status is invalid.
 	 */
 	public function set_status( ?string $status ): void {
+		$statuses = FulfillmentUtils::get_fulfillment_statuses();
+		if ( ! isset( $statuses[ $status ] ) ) {
+			throw new \InvalidArgumentException(
+				sprintf(
+				/* translators: %s is the invalid fulfillment status. */
+					esc_attr__( 'Invalid fulfillment status: %s', 'woocommerce' ),
+					esc_attr( $status )
+				)
+			);
+		}
+		// Set the fulfillment status.
+		$this->set_is_fulfilled( $statuses[ $status ]['is_fulfilled'] ?? false );
```

Comment:
just a small suggestion:

```suggestion
-		$this->set_is_fulfilled( $statuses[ $status ]['is_fulfilled'] ?? false );
+		$is_fulfilled = isset( $statuses[ $status ]['is_fulfilled'] ) ? $statuses[ $status ]['is_fulfilled'] : false;
+		$this->set_is_fulfilled( $is_fulfilled );
```